platform/linux-generic/odp_atomic.c

```diff
@@ -24,3 +26,210 @@ int odp_atomic_lock_free_u64(odp_atomic_op_t *atomic_op)
 	return 2;
 #endif
 }
+
+int odp_atomic_lock_free_u128(odp_atomic_op_t *atomic_op)
+{
+#ifdef _ODP_LOCK_FREE_128BIT_ATOMICS
+	if (atomic_op) {
+		atomic_op->all_bits = 0;
+		atomic_op->op.cas  = 1;
```

Comment:
The load and store operations are also lock-free.
```
atomic_op->op.load = 1;
atomic_op->op.store = 1;
atomic_op->op.cas = 1;
```

Reply:
Also return value should be 2, as all 128bit operations of the spec are lock-free (init is never lock-free, so it's not counted here).

Reply:
So far we have implemented cas, load and store operations.  Without supporting the rest in the list below, can we return 2 ?

struct {
                uint32_t init      : 1;  /**< Init atomic variable */
                uint32_t load      : 1;  /**< Atomic load */
                uint32_t store     : 1;  /**< Atomic store */
                uint32_t fetch_add : 1;  /**< Atomic fetch and add */
                uint32_t add       : 1;  /**< Atomic add */
                uint32_t fetch_sub : 1;  /**< Atomic fetch and subtract */
                uint32_t sub       : 1;  /**< Atomic subtract */
                uint32_t fetch_inc : 1;  /**< Atomic fetch and increment */
                uint32_t inc       : 1;  /**< Atomic increment */
                uint32_t fetch_dec : 1;  /**< Atomic fetch and decrement */
                uint32_t dec       : 1;  /**< Atomic decrement */
                uint32_t min       : 1;  /**< Atomic minimum */
                uint32_t max       : 1;  /**< Atomic maximum */
                uint32_t cas       : 1;  /**< Atomic compare and swap */
                uint32_t xchg      : 1;  /**< Atomic exchange */
        } op;

Reply:
Yes, since all 128 bit APIs of the spec are implemented as lock free.